includes/config/class-config.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * The configuration definition for HM Platform.
+ *
+ * @package hm-platform
+ */
+
+namespace HM\Platform\Config;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * Class Configuration
+ *
+ * @package HM\Platform\Config
+ */
+class Configuration implements ConfigurationInterface {
+	public function getConfigTreeBuilder() {
+		$tree_builder = new TreeBuilder();
+		$root_node = $tree_builder->root( 'plugins' );
+		$root_node
+			->children()
+				->arrayNode( 'infrastructure' )
+					->children()
+						->booleanNode( 'aws_ses_wp_mail' )
+							->defaultTrue()
+						->end()
+						->booleanNode( 'batcache' )
+							->defaultTrue()
+						->end()
+						->booleanNode( 'elasticpress' )
+							->defaultFalse()
+						->end()
+						->booleanNode( 'ludicrousdb' )
+							->defaultTrue()
+						->end()
+						->booleanNode( 'memcached' )
+							->defaultTrue()
+						->end()
+					->end()
+				->end()
+				->arrayNode( 'plugins' )
+					->children()
+						->booleanNode( 'bylines' )
+							->defaultFalse()
+						->end()
+						->booleanNode( 'cavalcade' )
+							->defaultTrue()
+						->end()
+						->booleanNode( 'hm-stack' )
```

Comment:
@joehoyle not formal yet but I will make a formal config definition part of my work next week. This is my original gist for thinking about what the config could be https://gist.github.com/roborourke/229fbc0ac0d90bf80e66e294eefc97dd

Reply:
I only included the `hm-stack` because it is included in the `$hm_platform` global. So the complicated convoluted abstraction started when someone added it there: https://github.com/humanmade/hm-platform/blob/beta/load.php#L125